api/client.go

```diff
@@ -53,6 +54,35 @@ func (c *Client) WalletBalance() (resp WalletBalance, err error) {
 	return
 }
 
+// Address returns an address controlled by the wallet.
+func (c *Client) WalletAddress() (resp Address, err error) {
+	err = c.get("/wallet/address", &resp)
+	return
+}
+
+// Addresses returns all addresses controlled by the wallet.
+func (c *Client) WalletAddresses() (resp Addresses, err error) {
+	err = c.get("/wallet/addresses", &resp)
+	return
+}
+
+// Transactions returns all transactions relevant to the wallet, ordered
+// oldest-to-newest.
+func (c *Client) Transactions() (resp Transactions, err error) {
+	err = c.get("/wallet/transactions", &resp)
+	return
+}
+
+func (c *Client) SendSiacoins(out types.SiacoinOutput) (resp SendReceipt, err error) {
+	err = c.post("/wallet/send/siacoins", out, &resp)
+	return
+}
+
+func (c *Client) BroadcastTransaction(txn types.Transaction) (resp SendReceipt, err error) {
+	err = c.post("/wallet/broadcast", txn, nil)
+	return
+}
```

Comment:
I wonder if we should "namespace" the client, i.e.
```go
type Client struct {
    Wallet WalletClient
    TxPool TxPoolClient
    // etc.
}
```
Then we could write `c.Wallet.Addresses()` instead of `c.WalletAddresses()`, and we wouldn't need to worry about name collisions. 🤔

Reply:
How likely/common are collisions? I guess we'll see as we add more endpoints

Reply:
It's nice because `Wallet.Transactions()` maps cleanly to the corresponding `/wallet/transactions` endpoint and `siac wallet transactions` command.

The main downside I can think of is that it means you can't define a single interface with all of the Client methods; instead, you'd have to do something like
```go
type APIClient interface {
    WalletAddress() (api.WalletAddress, error)
    WalletTransactions() (api.Transactions, error)
    TxpoolTransactions() (api.Transactions, error)
}

type clientWrapper struct {
    c *api.Client
}
func (cw clientWrapper) WalletAddress() (api.WalletAddress, error) {
    return cw.c.Wallet.Address()
}
func (cw clientWrapper) WalletTransactions() (api.Transactions, error) {
    return cw.c.Wallet.Transactions()
}
func (cw clientWrapper) TxpoolTransactions() (api.Transactions, error) {
    return cw.c.Txpool.Transactions()
}
```

You *could*, however, define interfaces for just `Wallet`, or just `Txpool`, etc. So this is only a problem if you're trying to mock the client as a whole.

Let's leave it for now and experiment with this design once the API is more fleshed out.